### PR TITLE
[BPK-1078] Flow support for native BpkIcon

### PR DESCRIPTION
**Pull request by shaundon**
Also added a `.flowconfig` to `/native` to make Flow play nicely with React Native. Comments about the code are inline.

### CODE REVIEW

File: native/.flowconfig [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[ignore]
```
Comment (shaundon) [inline native/.flowconfig]: This file is largely copied from a vanilla `create-react-native-app` app.

Comment (shaundon) [reply native/.flowconfig]: We may be able to remove bits but for now I've left it mostly intact.

File: native/.flowconfig [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+[options]
+emoji=true
+
+unsafe.enable_getters_and_setters=true
+
+module.file_ext=.js
+module.file_ext=.json
+module.file_ext=.ios.js
+module.file_ext=.android.js
```
Comment (shaundon) [inline native/.flowconfig]: Without these lines, we get errors from RN code that implicitly required `.ios` and `.android`, e.g.:

```
import { colorGray700 } from 'bpk-tokens/tokens/base.react.native';
```

File: native/packages/react-native-bpk-component-icon/src/BpkIcon.js [hunk 3]
```diff
@@ -42,7 +44,14 @@ const styles = StyleSheet.create({
 const mapCharacterCode = characterCode =>
   String.fromCharCode(parseInt(characterCode, 16));
 
-const BpkIcon = props => {
+type IconName = $Keys<typeof iconMappings>;
+type Props = {
+  icon: IconName,
+  small: ?boolean,
+  style: ?{} | Array<{}>,
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: This is probably going to become a common pattern, so we could abstract it out. Alternatively, we could pull a type out of `react-native` itself. For now, this works well.

File: native/packages/react-native-bpk-component-icon/stories.js [hunk 4]
```diff
@@ -16,6 +16,8 @@
  * limitations under the License.
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-icon/stories.js]: I tested that Flow works here by changing the `small` prop on an icon to a string (should be a boolean) and it does display an error.

File: native/packages/react-native-bpk-component-icon/src/BpkIcon.js [hunk 5]
```diff
@@ -42,7 +44,14 @@ const styles = StyleSheet.create({
 const mapCharacterCode = characterCode =>
   String.fromCharCode(parseInt(characterCode, 16));
 
-const BpkIcon = props => {
+type IconName = $Keys<typeof iconMappings>;
+type Props = {
+  icon: IconName,
+  small: ?boolean,
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: This shouldn't be optional, we always expect it to have a valid value (true or false).

File: native/packages/react-native-bpk-component-icon/src/BpkIcon.js (hunk 3 above)
Comment (k0nserv) [reply native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: How does React Native itself do these?

File: native/packages/react-native-bpk-component-icon/src/BpkIcon.js (hunk 5 above)
Comment (shaundon) [reply native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: But it *is* an optional proptype - that's what this is analogous to, right?

Comment (k0nserv) [reply native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: No that’s not true. The prop type is external to the component, but flow is internal i.e while a consumer can omit `small`, inside the component we expect it to always have a value. In this case React and the default props ensure that that is the case

Comment (shaundon) [reply native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: Ok, makes sense. I'll change it

**Pull request closed without merging**